### ISSUE TITLE
A simpler solution to the problem of focus?

### ISSUE THREAD

**Issue by captbaritone** [suda/tool-bar#235]
TL;DR: _There's another bug with handling focus when clicking on a toolbar button. I think there may be a simpler solution._

Tool Bar tries to behave as if the command that the toolbar button triggers was triggered from whatever had focus before the user clicked. As a result, Tool Bar actively gives focus to the DOM node that was focused before the user moved their mouse into the toolbar. Unfortunately giving a nodes can have side effects. For example, a pane will reveal itself when it receives focus.

This causes a bug, which can be reproduces by:

1. Have two Panels: PanelA and PanelB which are both in a visible dock
2. Focus PanelA
3. Move mouse to the toolbar (prevFocusedElm is now PanelA)
4. Click button which toggle's PanelB (PanelB is now focused)
5. Click button which toggle's PanelB (The dock containing PanelA is hidden)
6. Click button which toggle's PanelB

What happens now is:

1. prevFocusedElm (which is still PanelA) gets focused, causing the dock (and PanelB) to become visible
2. The command which toggles PanelB is now called with PanelB visible, so it closes PanelB instead of opening it

The net effect is that the third (and any subsequent) clicks in the toolbar do nothing, and the feature seems broken.

Proposed Solution
 =================

Rather than taking focus and then attempting to return it, what if wen never took that focus in the first place?

By calling `e.preventDefault` in a `mousedown` handler, we can avoid taking focus, and therefore we never have to return it. We may even be able to avoid tracking `prevFocusedElm` all together.

**If this solution seems reasonable, I'm happy to open a pull request.**

I think @AnWeber alluded to this issue on a comment here back in 2016: https://github.com/suda/tool-bar/pull/88

**Comment by captbaritone** [comment]
Ping @suda. This is affecting us on Nuclide.

**Comment by suda** [comment]
Hey, sorry for not responding to this issue! That sounds like a good way to go so feel free to create the PR!

**Comment by captbaritone** [comment]
@suda I've opened https://github.com/suda/tool-bar/pull/239. Let me know what you think.
